src/main/java/com/playmonumenta/networkchat/inlinereplacements/DiceRollReplacement.java

```diff
@@ -0,0 +1,74 @@
+package com.playmonumenta.networkchat.inlinereplacements;
+
+import com.playmonumenta.networkchat.utils.MMLog;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.TextDecoration;
+import org.apache.commons.lang3.tuple.Pair;
+import org.bukkit.command.CommandSender;
+import org.intellij.lang.annotations.RegExp;
+
+public class DiceRollReplacement extends InlineReplacement {
+
+	@RegExp
+	private static final String mDiceDataGivenRegex = ".* [1-9]?[0-9]*d[1-9][0-9]*.*";
```

Comment:
Ignoring for a moment that this should use https://docs.advntr.dev/minimessage/dynamic-replacements.html#complex-placeholders

This could be done more efficiently as:
```java
Random mRandom = new Random();
String mDiceDataGivenRegex = "(?<=^|[^\\])<roll ([1-9][0-9]*)d([1-9][0-9]*)>";
Pattern mDiceDataGivenPattern = Pattern.compile(mDiceDataGivenRegex, Pattern.CASE_INSENSITIVE);

// In the code below
filterResult.component().replaceText(matcher -> matcher.match(mDiceDataGivenPattern)
	.replacement((MatchResult match, TextComponent.Builder textBuilder) -> {
		MMLog.info("DRR: matched string '" + match.group() + "' with regex: " + mDiceDataGivenRegex);
		int numDice = Integer.parseInt(match.group(1));
		int dieFaces = Integer.parseInt(match.group(2));

		int rollValue = 0;
		for (int i = 0; i < numDice; i++) {
			rollValue += mRandom.nextInt(dieFaces) + 1;
		}
		textBuilder.append(Component.text("ROLL").decoration(TextDecoration.BOLD, true).hoverEvent(Component.text(rollValue + " out of " + numDice
			+ "d" + dieFaces + " (" + numDice * dieFaces + ")")));
		return textBuilder;
	}));
```
Regex explanation and proof, which adds support for an escaped version of this sequence: https://regex101.com/r/FDPfUJ/1

Reply:
And a snippet of what that would look like with the number before `d` being optional:
```regex
([1-9][0-9]*|)d
```
```java
String numDiceStr = match.group(1);
int numDice = numDiceStr.isEmpty() ? 1 : Integer.parseInt(numDiceStr);
```

Reply:
the number before d is already optional

Reply:
Yes, but this is currently matching the entire message, instead of just the `<roll #d#>` part, which requires extra work to identify the parts you want instead of using regex's built-in group matching. Anything in `()` gets assigned to a placeholder `$#`, where `$0` is the entire matched section (since `.*` is on either end, this is currently the whole message), and `$1` is the first section of text matched by parentheses.

Reply:
By putting the numbers you want in parentheses, you can be absolutely sure you've absolutely sure that `$1` (group 1) is no more or less than the number you wish to parse as an int.